app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue

```diff
@@ -227,14 +238,41 @@ export default {
     removeKeyboardListener() {
       document.removeEventListener("keydown", this.keyboardShortcutsListener);
     },
+    updateQueryName(value) {
+      const {
+        name: queryName
+      } = value;
+      this.labelValue = queryName
+      this.$root.$emit('enableSavedButton')
+    },
     onSaveEventHandler(opts) {
-      const { name } = opts
-      // if there's some name, shrink the other buttons
-      this.removeLabelBtn = !!name
 
+      const {
+        params: { queryId }
+      } = this.$route;
+
+      if (queryId) {
+        this.saveHandlerSavedQuery(opts)
+      } else {
+        this.saveHandlerNewQuery(opts)
+      }
+    },
+    saveHandlerSavedQuery(opts) {
       // send the query to be stored
       this.$root.$emit("storeCurrentQuery", opts);
     },
+    saveHandlerNewQuery(opts) {
```

Comment:
I don't like these methods: `onSaveEventHandler`, `saveHandlerSavedQuery`, `saveHandlerNewQuery`, what does SQLEditorHeader care about whether is a new or saved query? Me (SQLEditorHeader) emits: "hey, I'm saving the query". That's all.
My listener (Dataset, for sure) listens: "Are you saving? OK, that means the querySavingPrompt is visible, etc, etc..."